src/components/common/ModalComponent.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import { Modal } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import { SemanticToastContainer } from 'react-semantic-toasts';
+
+export default class ArtModal extends Component {
+  state = { modalOpen: false }
+
+  toggleModal = () => this.setState({ modalOpen: !this.state.modalOpen });
+
+  render() {
+    const { children } = this.props;
+    const childrenWithProps = React.Children.map(children, child =>
+      React.cloneElement(child,
+        { toggleModal: this.toggleModal }
+      )
+    );
+    return (
+      <div>
+        <Modal
+          trigger={
+            <i
+              className="plus link icon"
+              onClick={this.toggleModal}
+              onKeyUp={() => {}}
```

Comment:
Is this handler necessary when it does nothing?

Reply:
This is necessary to close the modal